src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryFileValidator.cs

```diff
@@ -0,0 +1,172 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Data;
+using System.Data.Common;
+using System.Data.SqlClient;
+using System.Globalization;
+using System.IO;
+using Microsoft.SqlServer.Management.Common;
+using Microsoft.SqlServer.Management.Sdk.Sfc;
+using Microsoft.SqlServer.Management.Smo;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.Connection.ReliableConnection;
+using Microsoft.SqlTools.ServiceLayer.FileBrowser;
+
+namespace Microsoft.SqlTools.ServiceLayer.DisasterRecovery
+{
+    /// <summary>
+    /// Validate selected file paths for backup/restore operations
+    /// </summary>
+    public static class DisasterRecoveryFileValidator
+    {
+        private const string localSqlServer = "(local)";
+        private const string localMachineName = ".";
+
+        public static bool ValidatePaths(FileBrowserValidateEventArgs args, out string errorMessage)
+        {
+            errorMessage = "";
+            bool result = true;
+            DbConnection dbConn = null;
+            ConnectionInfo connInfo;
+
+            if (args != null)
+            {
+                ConnectionService.Instance.TryFindConnection(args.OwnerUri, out connInfo);
+                SqlConnection conn = null;
+                ServerConnection serverConnection = null;
+                if (connInfo != null)
+                {
+                    connInfo.TryGetConnection(Connection.ConnectionType.Default, out dbConn);
+                    if (dbConn != null)
+                    {
+                        conn = ReliableConnectionHelper.GetAsSqlConnection(dbConn);
+                        if (conn != null)
+                        {
+                            serverConnection = new ServerConnection(conn);
+                        }
+                    }
+                }
+
+                if (serverConnection != null)
+                {
+                    bool isLocal = false;
+                    if (string.Compare(GetMachineName(serverConnection.ServerInstance), Environment.MachineName, StringComparison.OrdinalIgnoreCase) == 0)
+                    {
+                        isLocal = true;
+                    }
+
+                    foreach (string filePath in args.FilePaths)
+                    {
+                        bool IsFolder = false;
```

Comment:
>IsFolder [](start = 29, length = 8)

local variables should be camelCased..meaning first letter should be lower-case.  Please apply this rule to all code in the is PR.